shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -577,30 +400,21 @@ public boolean onHoverEvent(MotionEvent event) {
         return handled;
     }
 
+    /**
+     * Invoked by Android when a generic motion event occurs, e.g., joystick movement, mouse hover,
+     * track pad touches, scroll wheel movements, etc.
+     *
+     * Flutter handles all of its own gesture detection and processing, therefore this
+     * method forwards all {@link MotionEvent} data from Android to Flutter.
+     */
     @Override
     public boolean onGenericMotionEvent(MotionEvent event) {
-        // Method isFromSource is only available in API 18+ (Jelly Bean MR2)
-        // Mouse hover support is not implemented for API < 18.
-        boolean isPointerEvent = Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2
-            && event.isFromSource(InputDevice.SOURCE_CLASS_POINTER);
-        if (!isPointerEvent ||
-            event.getActionMasked() != MotionEvent.ACTION_HOVER_MOVE ||
-            !isAttached()) {
+        boolean handled = isAttached() && androidTouchProcessor.onGenericMotionEvent(event);
+        if (handled) {
```

Comment:
Couldn't you just do this?:
```suggestion
        return handled ? true : super.onGenericMotionEvent(event);
```
or get rid if the local variable and move the boolean into the if:
```suggestion
        if (isAttached() && androidTouchProcessor.onGenericMotionEvent(event)) {
```

Reply:
I went with your first option.